umbraco/Umbraco-CMS
# Adds bundle options to the package manifest

to more control over how bundling works for static file processing for app_plugins.

See https://github.com/umbraco/Umbraco-CMS/discussions/10469#discussioncomment-903273 for discussion.

The package.manifest has a new option: "bundleOptions:" The values can be:

* "Default" - which is the default value if none is specified and means that the package's assets will be processed within the main package bundle like it does today. (in debug, files will not be processed)
* "None" - which means that the assets in the package will not be processed at all and will all be requested as individual assets. This will essentially be a bundle that has composite processing turned off for both debug and production.
* "Independent" - which means the packages assets will be processed as it's own separate bundle. (in debug, files will not be processed)

## Other changes

* Added BundlingOptions - so that we can have more control over bundling behavior
* Simplifies the SmidgeRuntimeBundler with how it creates bundle options and base them on BundlingOptions
* Fixes CSS bundling to ensure that in debug mode we are using composite files

## Testing

* Create a couple App_Plugins with package.manifests declaring both CSS and JS (don't add the bundleOptions parameter yet). 
* Run umbraco and ensure those JS/CSS are loaded in both Debug and not Debug modes. In debug mode they will be their normal paths, in non-debug they will be bundled.
* Change the package manifest to include "bundleOptions": "None"
* Run umbraco and ensure those JS/CSS are loaded in both Debug and not Debug modes. In both modes, the assets will be loaded as their original paths but they will also include the cache buster query string
* Change the package manifest to include "bundleOptions": "Independent"
* Run umbraco and ensure those JS/CSS are loaded in both Debug and not Debug modes. In debug mode they will be their normal paths, in non-debug they will be bundled, but there will be a separate bundle for each package.
